cypress/integration/team/teams_spec.js

```diff
@@ -107,4 +107,36 @@ describe('Teams Suite', () => {
         // 9. Remove user from team
         cy.removeTeamMember(teamURL, user.firstName);
     });
+
+    it('TS14633 Leave all teams', () => {
+        // 1. Login and go to /
+        cy.apiLogin('user-2');
+        cy.visit('/');
+
+        function leaveTeamUntilCant() {
+            cy.leaveTeam();
+            // eslint-disable-next-line max-nested-callbacks
+            cy.apiGetTeams().then((response) => {
+                if (response.body.length > 0) {
+                    leaveTeamUntilCant();
+                }
+            });
+        }
+
+        // * check for user not being on any teams
+        // * if they are on atleast 1, start leaving teams
+        // eslint-disable-next-line max-nested-callbacks
+        cy.apiGetTeams().then((response) => {
+            if (response.body.length > 0) {
+                leaveTeamUntilCant();
+            }
+        });
```

Comment:
[loc 116-133](https://github.com/mattermost/mattermost-webapp/blob/42dd6d7dd75c8235c1015efd8ae5176e92dfa500/cypress/integration/team/teams_spec.js#L116-L133) is a bit repetitive.  Can we simplify it like,
```javascript
function leaveAllTeams() {
    cy.apiGetTeams().then((response) => {
        if (response.body.length > 0) {
            cy.leaveTeam().then(() => leaveAllTeams());
        }
    });
}
leaveAllTeams();
```
Also, you may add `/*eslint max-nested-callbacks: ["error", 5]*/` on top of this file to fix ESLint warning